File: java/src/main/java/model/Partida.java
package model;

import java.time.LocalDate;

public class Partida {
    private int ano;
    private int rodada;
    private LocalDate data;
    private String mandante;
    private String visitante;
    private int golsMandante;
    private int golsVisitante;
    
    public Partida(int ano, int rodada, LocalDate data, String mandante, String visitante, 
                   int golsMandante, int golsVisitante) {
        this.ano = ano;
        this.rodada = rodada;
        this.data = data;
        this.mandante = mandante;
        this.visitante = visitante;
        this.golsMandante = golsMandante;
        this.golsVisitante = golsVisitante;
    }
    
    public boolean contemTime(String nomeTime) {
        return mandante.equals(nomeTime) || visitante.equals(nomeTime);
    }
    
    public String getResultadoPara(String nomeTime) {
        if (mandante.equals(nomeTime)) {
            if (golsMandante > golsVisitante) return "V";
            else if (golsMandante == golsVisitante) return "E";
            else return "D";
        } else if (visitante.equals(nomeTime)) {
            if (golsVisitante > golsMandante) return "V";
            else if (golsVisitante == golsMandante) return "E";
            else return "D";
        }
        return "";
    }
    
    public int getGolsFeitosPor(String nomeTime) {
        if (mandante.equals(nomeTime)) return golsMandante;
        else if (visitante.equals(nomeTime)) return golsVisitante;
        return 0;
    }
    
    public int getGolsSofridosPor(String nomeTime) {
        if (mandante.equals(nomeTime)) return golsVisitante;
        else if (visitante.equals(nomeTime)) return golsMandante;
        return 0;
    }
    
    // Getters
    public int getAno() { return ano; }
    public int getRodada() { return rodada; }
    public LocalDate getData() { return data; }
    public String getMandante() { return mandante; }
    public String getVisitante() { return visitante; }
    public int getGolsMandante() { return golsMandante; }
    public int getGolsVisitante() { return golsVisitante; }
}

File: python/model/partida.py
from datetime import date

class Partida:
    def __init__(self, ano: int, rodada: int, data: date, mandante: str, 
                 visitante: str, gols_mandante: int, gols_visitante: int):
        self.ano = ano
        self.rodada = rodada
        self.data = data
        self.mandante = mandante
        self.visitante = visitante
        self.gols_mandante = gols_mandante
        self.gols_visitante = gols_visitante
    
    def contem_time(self, nome_time: str) -> bool:
        """Verifica se o time participa desta partida"""
        return self.mandante == nome_time or self.visitante == nome_time
    
    def get_resultado_para(self, nome_time: str) -> str:
        """Retorna V, E ou D para o time especificado"""
        if self.mandante == nome_time:
            if self.gols_mandante > self.gols_visitante:
                return "V"
            elif self.gols_mandante == self.gols_visitante:
                return "E"
            else:
                return "D"
        elif self.visitante == nome_time:
            if self.gols_visitante > self.gols_mandante:
                return "V"
            elif self.gols_visitante == self.gols_mandante:
                return "E"
            else:
                return "D"
        return ""
    
    def get_gols_feitos_por(self, nome_time: str) -> int:
        """Retorna gols feitos pelo time"""
        if self.mandante == nome_time:
            return self.gols_mandante
        elif self.visitante == nome_time:
            return self.gols_visitante
        return 0
    
    def get_gols_sofridos_por(self, nome_time: str) -> int:
        """Retorna gols sofridos pelo time"""
        if self.mandante == nome_time:
            return self.gols_visitante
        elif self.visitante == nome_time:
            return self.gols_mandante
        return 0
